File: smoke_tests/csharp/weak_crypto.cs
// Weak Cryptography vulnerabilities in C#
using System;
using System.Security.Cryptography;
using System.Text;

public class WeakCryptoVulnerabilities
{
    public string HashMd5(string input)
    {
        // VULNERABLE: MD5 is cryptographically broken
        using (MD5 md5 = MD5.Create())
        {
            byte[] hash = md5.ComputeHash(Encoding.UTF8.GetBytes(input));
            return BitConverter.ToString(hash);
        }
    }

    public string HashSha1(string input)
    {
        // VULNERABLE: SHA1 is deprecated
        using (SHA1 sha1 = SHA1.Create())
        {
            byte[] hash = sha1.ComputeHash(Encoding.UTF8.GetBytes(input));
            return BitConverter.ToString(hash);
        }
    }

    public byte[] EncryptDes(byte[] data, byte[] key)
    {
        // VULNERABLE: DES is obsolete
        using (DES des = DES.Create())
        {
            des.Key = key;
            des.Mode = CipherMode.ECB; // VULNERABLE: ECB mode
            return des.CreateEncryptor().TransformFinalBlock(data, 0, data.Length);
        }
    }

    public int GenerateToken()
    {
        // VULNERABLE: Non-cryptographic random
        Random rand = new Random();
        return rand.Next();
    }

    public string WeakSessionId()
    {
        // VULNERABLE: Predictable session ID
        return DateTime.Now.Ticks.ToString();
    }
}

File: smoke_tests/kotlin/weak_crypto.kt
// Weak Cryptography vulnerabilities in Kotlin
import java.security.MessageDigest
import javax.crypto.Cipher
import javax.crypto.spec.SecretKeySpec
import java.util.Random

class WeakCryptoVulnerabilities {
    fun hashMd5(input: String): String {
        // VULNERABLE: MD5 is cryptographically broken
        val md = MessageDigest.getInstance("MD5")
        val digest = md.digest(input.toByteArray())
        return digest.joinToString("") { "%02x".format(it) }
    }

    fun hashSha1(input: String): String {
        // VULNERABLE: SHA1 is deprecated
        val md = MessageDigest.getInstance("SHA-1")
        val digest = md.digest(input.toByteArray())
        return digest.joinToString("") { "%02x".format(it) }
    }

    fun encryptDes(data: ByteArray, key: ByteArray): ByteArray {
        // VULNERABLE: DES is obsolete
        val secretKey = SecretKeySpec(key, "DES")
        val cipher = Cipher.getInstance("DES/ECB/PKCS5Padding")
        cipher.init(Cipher.ENCRYPT_MODE, secretKey)
        return cipher.doFinal(data)
    }

    fun generateToken(): Int {
        // VULNERABLE: Non-cryptographic random
        return Random().nextInt(1000000)
    }

    fun weakSessionId(): String {
        // VULNERABLE: Predictable session ID
        return System.currentTimeMillis().toString()
    }

    fun ecbEncryption(data: ByteArray, key: ByteArray): ByteArray {
        // VULNERABLE: ECB mode
        val cipher = Cipher.getInstance("AES/ECB/PKCS5Padding")
        cipher.init(Cipher.ENCRYPT_MODE, SecretKeySpec(key, "AES"))
        return cipher.doFinal(data)
    }
}
